src/link/http/createHttpLink.ts

```diff
@@ -129,17 +130,31 @@ export const createHttpLink = (linkOptions: HttpOptions = {}) => {
     const definitionIsMutation = (d: DefinitionNode) => {
       return d.kind === 'OperationDefinition' && d.operation === 'mutation';
     };
+    const definitionIsSubscription = (d: DefinitionNode) => {
+      return d.kind === 'OperationDefinition' && d.operation === 'subscription';
+    };
+    const isSubscription = definitionIsSubscription(getMainDefinition(operation.query));
+    // does not match custom directives beginning with @defer
+    const hasDefer = hasDirectives(['defer'], operation.query);
     if (
       useGETForQueries &&
       !operation.query.definitions.some(definitionIsMutation)
     ) {
       options.method = 'GET';
     }
 
-    // does not match custom directives beginning with @defer
-    if (hasDirectives(['defer'], operation.query)) {
+    if (hasDefer || isSubscription) {
       options.headers = options.headers || {};
-      options.headers.accept = "multipart/mixed; deferSpec=20220824, application/json";
+      let acceptHeader = "multipart/mixed;";
+      // Omit defer-specific headers if the user attempts to defer a selection
+      // set on a subscription and log a warning.
+      if (isSubscription && hasDefer) {
+        invariant.warn("Multipart-subscriptions do not support @defer");
+      }
+
+      if (isSubscription) acceptHeader += 'boundary=graphql;subscriptionSpec=1.0,application/json';
+      if (!isSubscription && hasDefer) acceptHeader += 'deferSpec=20220824,application/json';
```

Comment:
```suggestion
      if (isSubscription) {
        acceptHeader += 'boundary=graphql;subscriptionSpec=1.0,application/json';
      } else if (hasDefer) {
        acceptHeader += 'deferSpec=20220824,application/json';
      }
```

[nit] You might consider an `else if` here to avoid needing to repeat `isSubscription`.